x-pack/plugins/security/public/authorization/authorization_service.ts

```diff
@@ -5,18 +5,30 @@
  * 2.0.
  */
 
+import type { HttpSetup } from '@kbn/core/public';
 import type { AuthorizationServiceSetup } from '@kbn/security-plugin-types-public';
+import type { AuthorizationCurrentUserApiKeyPrivilegesResponse } from '@kbn/security-plugin-types-public/src/authorization/authorization_service';
 
 import type { ConfigType } from '../config';
 
 interface SetupParams {
   config: ConfigType;
+  http: HttpSetup;
 }
 
 export class AuthorizationService {
-  public setup({ config }: SetupParams): AuthorizationServiceSetup {
+  public setup({ config, http }: SetupParams): AuthorizationServiceSetup {
     const isRoleManagementEnabled = () => config.roleManagementEnabled;
 
-    return { isRoleManagementEnabled };
+    const getCurrentUserApiKeyPrivileges = async () => {
+      const { canManageApiKeys, canManageCrossClusterApiKeys, canManageOwnApiKeys } =
+        (await http.get(
+          '/internal/security/api_key'
+        )) as AuthorizationCurrentUserApiKeyPrivilegesResponse;
+
+      return { canManageApiKeys, canManageCrossClusterApiKeys, canManageOwnApiKeys };
+    };
+
+    return { isRoleManagementEnabled, getCurrentUserApiKeyPrivileges };
```

Comment:
`GET /internal/security/api_key` returns all API keys, which if there are many could make this slow. Additionally, this endpoint is being dropped and replaced with a query endpoint. See https://github.com/elastic/kibana/pull/168970/

I'd suggest instead augmenting our authentication service to include the functionality that you want. This is where we already expose API key functions. See https://github.com/elastic/kibana/blob/bcbd55088ccacc9e873675948b02b3b522969c20/x-pack/plugins/security/public/authentication/authentication_service.ts#L44

You could add a `getCurrentUserApiKeyPrivileges` function in the authentication service that calls a new internal endpoint that ONLY returns the three flags you are looking for (canManageApiKeys, canManageCrossClusterApiKeys, canManageOwnApiKeys), e.g. `GET /internal/security/api_key/_check_privileges`

Additionally, this would be exposed on the server side as well here: https://github.com/elastic/kibana/blob/ba8234e6ee081690f0f189ab980e3cef6e4f8e1b/x-pack/plugins/security/server/authentication/authentication_service.ts#L387